src/main/java/com/dongsoop/dongsoop/chat/validator/ChatValidator.java

```diff
@@ -31,160 +29,169 @@ public ChatValidator(@Qualifier("redisChatRepository") ChatRepository chatReposi
+        boolean isSelfChat = user1.equals(user2);
+        throwExceptionIfSelfChat(isSelfChat);
     }
 
-    private void handleParticipantAddition(ChatRoom room, Long userId) {
-        boolean shouldAddUser = shouldAddUserToRoom(userId, room);
-        if (shouldAddUser) {
-            addUserToRoom(room, userId);
+    private void throwExceptionIfSelfChat(boolean isSelfChat) {
+        if (isSelfChat) {
+            throw new SelfChatException();
         }
     }
 
-    private boolean shouldAddUserToRoom(Long userId, ChatRoom room) {
-        return !ANONYMOUS_USER_ID.equals(userId) && !room.getParticipants().contains(userId);
+    private void validateMessageRequirements(ChatMessage message) {
+        boolean hasRequiredFields = hasRequiredFields(message);
+        throwExceptionIfInvalidMessage(hasRequiredFields);
     }
 
-    private void addUserToRoom(ChatRoom room, Long userId) {
-        room.getParticipants().add(userId);
-        chatRepository.saveRoom(room);
+    private boolean hasRequiredFields(ChatMessage message) {
+        boolean hasRoomId = StringUtils.hasText(message.getRoomId());
+        boolean hasSenderId = message.getSenderId() != null;
+        return hasRoomId && hasSenderId;
     }
 
-    private void validateMessageRequirements(ChatMessage message) {
-        boolean hasRequiredFields = hasRequiredFields(message);
-        if (!hasRequiredFields) {
+    private void throwExceptionIfInvalidMessage(boolean hasRequiredFields) {
+        boolean isInvalid = !hasRequiredFields;
+        if (isInvalid) {
             throw new InvalidChatRequestException();
         }
     }
 
-    private boolean hasRequiredFields(ChatMessage message) {
-        return StringUtils.hasText(message.getRoomId()) && message.getSenderId() != null;
+    private void enrichMessageId(ChatMessage message) {
+        boolean messageIdIsNull = message.getMessageId() == null;
+        setMessageIdIfNull(messageIdIsNull, message);
     }
 
-    private ChatMessage enrichMessageData(ChatMessage message) {
-        setMessageIdIfAbsent(message);
-        setTimestampIfAbsent(message);
-        setMessageTypeIfAbsent(message);
-        return message;
-    }
-
-    private void setMessageIdIfAbsent(ChatMessage message) {
-        if (message.getMessageId() == null) {
+    private void setMessageIdIfNull(boolean messageIdIsNull, ChatMessage message) {
+        if (messageIdIsNull) {
             message.setMessageId(UUID.randomUUID().toString());
         }
     }
 
-    private void setTimestampIfAbsent(ChatMessage message) {
-        if (message.getTimestamp() == null) {
+    private void enrichTimestamp(ChatMessage message) {
+        boolean timestampIsNull = message.getTimestamp() == null;
+        setTimestampIfNull(timestampIsNull, message);
+    }
+
+    private void setTimestampIfNull(boolean timestampIsNull, ChatMessage message) {
+        if (timestampIsNull) {
             message.setTimestamp(LocalDateTime.now());
         }
     }
```

Comment:
불필요한 메서드 분리로 보여집니다.
```suggestion
    private void enrichTimestamp(ChatMessage message) {
        if (message.getTimestamp() == null) {
            message.setTimestamp(LocalDateTime.now());
        }
    }
```

객체지향적으로 설계한다면 message 내부에 구현해도 될 것 같아 보입니다.

```suggestion
    private void setTimestamp(LocalDateTime timestamp) {
        if (this.timestamp() == null) {
            this.timestamp = timestamp;
        }
    }
```